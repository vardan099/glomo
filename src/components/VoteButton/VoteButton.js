import React from 'react';
import PropTypes from "prop-types";

const VoteButton = (props) => {
    const {buttonText, clickHandler, voteType, gameId, type} = props;
    return (
        <button data-test="voteButton" className={`vote-button bg-${type}`} onClick={() => clickHandler(gameId, voteType)}>{buttonText}</button>
    )
};

VoteButton.propTypes = {
    buttonText: PropTypes.string,
    clickHandler: PropTypes.func,
    voteType: PropTypes.number,
    type: PropTypes.string,
    gameId: PropTypes.string
};


export default VoteButton;