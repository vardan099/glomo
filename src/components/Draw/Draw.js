import React from 'react';
import PropTypes from "prop-types";
import VoteButton from "../VoteButton/VoteButton";
import vs from '../../images/VS.png'

const Draw = (props) => {
    const {country, state, voteHandler, objectId,sport, type, group} = props;
    return (
        <div className="sports-poll_draw">
            <h3>{group} - {sport}</h3>
            <img src={vs} alt="vs"/>
            <h4><span className="info-label">Country:</span>{country}</h4>
            <h4><span className="info-label">Status:</span>{state}</h4>
            <VoteButton
                buttonText="Draw"
                clickHandler={voteHandler}
                voteType={2}
                gameId={objectId}
                type={type}
            />
        </div>
    )
};

Draw.propTypes = {
    country: PropTypes.string,
    state: PropTypes.string,
    sport: PropTypes.string,
    group: PropTypes.string,
    voteHandler: PropTypes.func,
    type: PropTypes.string,
    objectId: PropTypes.string
};


export default Draw;