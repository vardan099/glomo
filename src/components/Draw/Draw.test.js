import React from 'react';
import { shallow } from 'enzyme';
import { findByTestAtrr, checkProps } from '../../TestUtils/index';
import Draw from './Draw'
import VoteButton from '../VoteButton/VoteButton'

describe('Draw Component', () => {

    describe('Checking PropTypes', () => {
        it('Should NOT throw a warning', () => {
            const expectedProps = {
                country: jest.string,
                state: jest.string,
                sport: jest.string,
                group: jest.string,
                voteHandler: jest.func,
                type: jest.string,
                objectId: jest.string
            };
            const propsError = checkProps(Draw, expectedProps);
            expect(propsError).toBeUndefined();
        });

    });

    describe('Component Renders', () => {
        let wrapper;
        beforeEach(() => {
            const props = {
                country: "Sweden",
                state: "STARTED",
                sport: "FOOTBALL",
                group: jest.string,
                voteHandler: ()=>{},
                type: "draw",
                objectId: "fZZUhitsVt"
            };
            wrapper = shallow(<Draw {...props} />);
        });

        it('renders correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('Should render VoteButton Component', () => {
            const desc = wrapper.find(VoteButton);
            expect(desc.length).toBe(1);
        });

    });




});