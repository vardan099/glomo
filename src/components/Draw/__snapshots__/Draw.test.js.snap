// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draw Component Component Renders renders correctly 1`] = `ShallowWrapper {}`;
